src/app/next-game-prediction/next-game-prediction.component.ts

```diff
@@ -0,0 +1,62 @@
+import { Component, OnInit, OnChanges, Input } from '@angular/core';
+
+import { TipService } from '../tip.service';
+import { Team } from '../team';
+import { Tip } from '../tip';
+
+@Component({
+  selector: 'app-next-game-prediction',
+  templateUrl: './next-game-prediction.component.html',
+  styleUrls: ['./next-game-prediction.component.css']
+})
+export class NextGamePredictionComponent implements OnInit, OnChanges {
+  @Input() team: Team;
+  otherTeam: Team;
+  @Input() teams: Team[];
+  latestTip: Tip;
+
+  constructor(private tipService: TipService) { }
+
+  getTeam(teamId: number): Team {
+    return this.teams.find(team => (team.id === teamId));
+  }
+
+  getTeamUrl(teamId: number) {
+    return `https://squiggle.com.au/${this.getTeam(teamId).logo}`;
+  }
+
+  getTeamChance() {
+    if (this.team.id === this.latestTip.ateamid) {
+      return this.latestTip.confidence;
+    } else {
+      return this.latestTip.hconfidence;
+    }
+  }
+
+  ngOnInit(): void {
+    this.getLatestTip();
+  }
+
+  ngOnChanges(): void {
```

Comment:
```suggestion
  ngOnChanges(): void {
    this.getLatestTip();
  }
```
Do you not have to add an if statement to check when changes are made to either of the `@Input`s? Or does it run fine without it?

Reply:
My understanding was that it would only call this if the input parameters change. If it gets updated more frequently, then we might send multiple requests, but I haven't seen this behavior in the network monitor on Firefox, during my testing, so it should be fine.